### DISCUSSION TITLE
SSG build error: no prerender() hook returned (an) matching URL(s)

### DISCUSSION THREAD

**Discussion by spryzone** [vikejs/vike#1252] [Discussion] [answered]
I am receiving this error in the console when I try to build my SSG site: 

`[vike][Warning] Cannot pre-render page defined at /pages/products/@id because it has a non-static route, and no prerender() hook returned (an) URL(s) matching the page's route.`

I had a website (SPA using react-router) which I have converted to Vike (V1 architecture) so I get HTML pages for each route. The content for the pages comes from a static JSON file. Everything looks good in dev (no 404s). In the `dist` folder, I am getting HTML pages for the static routes, but not for the dynamic routes. I followed the docs and as far as I can tell, the only difference is that my content is static, and the docs Star Wars example is fetching data from an API. 

**Structure**

```
- pages
    |__services
             |__@id
                  |__+onBeforeRender.ts
                  |__+Page.tsx

```

**+onBeforeRender.ts**
```javascript
import type { PageContextBuiltInServer } from 'vike/types';
import content from '../../../content.json';
import { findDoc } from '@/components/helpers/findDoc';

export default async function onBeforeRender(pageContext: PageContextBuiltInServer) {
  let page = await findDoc(content, 'sectors', pageContext.routeParams.id);
  return {
    pageContext: {
      pageProps: {
        page,
      },
      title: page?.title,
    },
  };
}
```

**config.h.ts**
```javascript
import type { Config } from 'vike/types';
import Layout from '../layouts/LayoutDefault';
import Head from '../layouts/HeadDefault';
import logoUrl from '/favicon.png';
import vikeReact from 'vike-react';

// Default config (can be overridden by pages)
export default {
  Layout,
  Head,
  title: 'Austere Risk',
  favicon: logoUrl,
  extends: vikeReact,
} satisfies Config;
```

**vite.config.ts**
```javascript
import ssr from 'vike/plugin';
import react from '@vitejs/plugin-react';
import { defineConfig } from 'vite';
import path from 'path';

export default defineConfig({
  plugins: [
    react({
      jsxRuntime: 'classic',
    }),
    ssr({ prerender: true }),
  ],
  optimizeDeps: { include: ['react/jsx-runtime', 'vike-react/renderer/onRenderClient'] },
  resolve: {
    alias: {
      '$/': path.resolve(__dirname, './'),
      '@/components': path.resolve(__dirname, '../../components'),
    },
  },
});
```

**package.json**
```json
    "vike": "^0.4.144",
    "vike-react": "^0.3.0"
```

Appreciate any suggestions where I might be going wrong.

**Comment by brillout** [comment]
It seems to be a user land issue, but minimal reproduction welcome if you believe otherwise.

**Comment by brillout** [comment, marked as answer]
Actually, the warning is misleading, I just fixed it: ca10f32cd05c50d001c6acf6904deec543636bfa.

Make sure to rename your `prerender()` hooks to https://vike.dev/onBeforePrerenderStart.

**Reply by spryzone** [reply]
Great, thanks for the feedback and fix!

**Reply by brillout** [reply]
Sure, btw, would your company be up for [sponsoring](https://github.com/sponsors/brillout)?
